File: backend/src/main/java/project/backend/domain/member/dto/PasswordChangeRequest.java
package project.backend.domain.member.dto;

import jakarta.validation.constraints.AssertTrue;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Getter;

@Getter
public class PasswordChangeRequest {

	@NotBlank(message = "현재 비밀번호를 입력해주세요.")
	String currentPassword;

	@NotBlank(message = "새로운 비밀번호를 입력해주세요.")
	@Size(min = 4, message = "비밀번호는 최소 4자 이상이여야 합니다.")
	String newPassword;

	@NotBlank(message = "새로운 비밀번호 확인을 입력해주세요.")
	@Size(min = 4, message = "비밀번호는 최소 4자 이상이여야 합니다.")
	String confirmPassword;

	@AssertTrue(message = "비밀번호와 확인값이 일치하지 않습니다.")
	boolean isPasswordMatch() {
		return newPassword.equals(confirmPassword);
	}

}

File: backend/src/main/java/project/backend/domain/member/dto/PasswordChangeRequest.kt
package project.backend.domain.member.dto

import jakarta.validation.constraints.AssertTrue
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class PasswordChangeRequest(

    @field:NotBlank(message = "현재 비밀번호를 입력해주세요.")
    val currentPassword: String,

    @field:NotBlank(message = "새로운 비밀번호를 입력해주세요.")
    @field:Size(min = 4, message = "비밀번호는 최소 4자 이상이여야 합니다.")
    val newPassword: String,

    @field:NotBlank(message = "새로운 비밀번호 확인을 입력해주세요.")
    @field:Size(min = 4, message = "비밀번호는 최소 4자 이상이여야 합니다.")
    val confirmPassword: String

) {
    @get:AssertTrue(message = "비밀번호와 확인값이 일치하지 않습니다.")
    val isPasswordMatch: Boolean
        get() = newPassword == confirmPassword
}